torch/_inductor/fx_passes/quantization.py

```diff
@@ -1470,15 +1470,24 @@ def woq(match: Match, *args, **kwargs):
         scales = kwargs["scales"]
         counters["inductor"]["woq_matcher_count"] += 1
         counters["inductor"]["woq_matcher_nodes"] += len(match.nodes)
-        out_features = weight.get_size()[0]
-        origin_x_size = x.get_size()
-        x_shape = [-1, origin_x_size[-1]]
-        out_shape = origin_x_size[:-1] + [
-            out_features,
-        ]
-        func1 = L[computation_reshape](x, x_shape)
-        func2 = L[computation_woq](func1, weight, scales)
-        return L[computation_reshape](func2, out_shape)
+        scales_dtype = scales.dtype
+        if scales_dtype == torch.float32:
+            scales = L[prims.convert_element_type.default](scales, torch.bfloat16)
+        if computation_reshape:
+            out_features = weight.get_size()[0]
+            origin_x_size = x.get_size()
+            x_shape = [-1, origin_x_size[-1]]
+            out_shape = origin_x_size[:-1] + [
+                out_features,
+            ]
+            func1 = L[computation_reshape](x, x_shape)
+            func2 = L[computation_woq](func1, weight, scales)
```

Comment:
I'm a bit confused about the output dtype of woq. Here, we explicitly convert scales to the dtype of the input tensor to match the dtypes and then convert it back to the dtype of scales to match the dtype of the original subgraph. But would these down-conversation worsen the accuracy? Instead, is it better to make the woq return the dtype that matches the scales directly?